src/Analysis/Codelyzer.Analysis.Common/FileUtils.cs

```diff
@@ -223,6 +243,37 @@ public static IEnumerable<string> GetProjectCodeFiles(string projectFile, string
                 return false; 
             }
         }
+
+        private static List<string> GetSharedProjectCodeFile(string sharedProjPath)
+        {
+            var documents = new List<string>();
+            try
+            {
+                var sharedDoc = new XmlDocument();
+                sharedDoc.Load(sharedProjPath);
+                var elements = sharedDoc.GetElementsByTagName("Compile");
+                var directory = Path.GetDirectoryName(sharedProjPath);
+
+                foreach (XmlNode element in elements)
+                {
+                    string relativePath = element.Attributes?["Include"]?.Value ?? "";
```

Comment:
If `"Include"` does not exist in `element.Attributes`, will `element.Attributes?["Include"]?` return null or will it throw an error?

Reply:
i haven't seen any code file element without "Include" attribute yet.   But it's a good check.  I will add element.Attributes != null first .